### ISSUE TITLE
[Bug] - error in attention mask LM tutorial

### ISSUE THREAD

**Issue by alexanderwerning** [pytorch/tutorials#2250]
### Add Link

https://pytorch.org/tutorials/beginner/transformer_tutorial.html

### Describe the bug

While following along the LM tutorial, I noticed that the attention mask is incorrect, as the diagonal is not zeros (which the docstring also does state), but ones.

line 95 to 97 in `tutorials/beginner_source/transformer_tutorial.py1`:
```
def generate_square_subsequent_mask(sz: int) -> Tensor:
    """Generates an upper-triangular matrix of -inf, with zeros on diag."""
    return torch.triu(torch.ones(sz, sz) * float('-inf'), diagonal=1)  # <-- should be diagonal=0?
```

### Describe your environment

N/A

cc @suraj813
